yrs/src/types/xml.rs

```diff
@@ -938,21 +938,29 @@ where
 
     /// Tree walker used depth-first search to move over the xml tree.
     fn next(&mut self) -> Option<Self::Item> {
+        fn try_descend(item: &Item) -> Option<&Block> {
+            if let ItemContent::Type(t) = &item.content {
+                let inner = t.as_ref();
+                let type_ref = inner.type_ref();
+                if !item.is_deleted()
+                    && (type_ref == TYPE_REFS_XML_ELEMENT || type_ref == TYPE_REFS_XML_FRAGMENT)
+                {
+                    return inner.start.as_deref();
+                }
+            }
+
+            None
+        }
+
         let mut result = None;
         let mut n = self.current.take();
         if let Some(current) = n {
             if !self.first_call || current.is_deleted() {
                 while {
-                    if let ItemContent::Type(t) = &current.content {
```

Comment:
Original source of an issue: we only alter node in while loop if its content type was `Type`. If it was i.e. `Deleted` we never updated the node nor break from the loop. The correct behavior is to descend on non-deleted `Type`, otherwise try to move right, then up to a parent hierarchy, even if current node was deleted.